tests/integrationv2/providers.py

```diff
@@ -49,6 +50,18 @@ def setup_server(self):
         """
         raise NotImplementedError
 
+    @classmethod
+    def supports_max_frag(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def supports_protocol(cls, protocol, with_cert=None):
+        raise NotImplementedError
+
+    @classmethod
+    def supports_cipher(cls, cipher, with_curve=None):
+        raise NotImplementedError
+
```

Comment:
We may need a method for supports_curves/supports_groups as well here.

Reply:
Possibly. When those use cases arrive we can easily add them.